.github/workflows/testdriver.yml

```diff
@@ -100,8 +100,7 @@ jobs:
       - name: Check out current branch
         uses: actions/checkout@v3
         with:
-          ref: ${{ github.head_ref }}
-
+          ref: main
```

Comment:
Based on https://github.com/actions/checkout?tab=readme-ov-file#usage, we could remove the `with` entirely (which defaults to the default branch)